lib/minitest/mock.rb

```diff
@@ -222,8 +227,10 @@ def stub name, val_or_callable, *block_args
 
     yield self
   ensure
-    metaclass.send :undef_method, name
-    metaclass.send :alias_method, name, new_name
-    metaclass.send :undef_method, new_name
+    unless metaclass.nil?
```

Comment:
@zamith, forget what I said about the "Undo", It is exactly what the current `ensure` block is doing.

I actually can make the tests pass with this at the end instead of `ensure`:

```ruby
out = yield self
metaclass.send :undef_method, name
metaclass.send :alias_method, name, new_name
metaclass.send :undef_method, new_name
out
```

Although it made me realise that while this passes, it just means that we can only test for what we predict. The `ensure` block is definitely a reassurance, although we have no real test which requires the `ensure` block.

Now I'm gonna **ensure** I have a cup of coffee.